packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol

```diff
@@ -461,9 +506,31 @@ contract BridgePool is Testable, BridgePoolInterface, ExpandedERC20 {
     }
 
     /**************************************
-     *        INTERNAL FUNCTIONS          *
+     *    INTERNAL & PRIVATE FUNCTIONS    *
      **************************************/
 
+    // Update the internal fee counter metrics by adding in any accumulated fees from the last time this logic was
+    // called. Considers any allocated fees that are assigned to the LPs at the conclusion of a bridging action.
+    function updateFeeCounters(uint256 allocatedLpFees) internal {
+        // Calculate the unallocatedAccumulatedFees from the last time the contract was called.
+        uint256 unallocatedAccumulatedFees = getUnallocatedAccumulatedFees();
+
+        // Decrement the undistributedLpFees by the amount of accumulated fees.
+        undistributedLpFees = undistributedLpFees > unallocatedAccumulatedFees
+            ? undistributedLpFees - unallocatedAccumulatedFees
+            : 0;
+
+        lastLpFeeUpdate = getCurrentTime();
+
+        // If this method was called from settleRelay then it includes the exact amount of fees to be attributed to the
+        // Liquidity providers. add this to the total undistributed LP fees and the utalized reserves. Adding it to the
+        // utalized reserves acts to book keep the fees while they are in transit.
+        if (allocatedLpFees > 0) {
+            undistributedLpFees += allocatedLpFees;
+            utilizedReserves += int256(allocatedLpFees);
```

Comment:
Note: we should be careful about casting. I think casting _makes sense_, but we need to be very very sure that it's not possible to pass in a huge number (> 2^255) here (i.e. that must _always revert_, it being disputable is not enough).

Open question whether we should always do safe casting where we revert if the uint is too large.

Reply:
agreed we need caution in this case. The only situation I can think where this would be an issue is when the `amount` transferred (and therefore the `allocatedLpFees` is absolutely massive (> 2^255). this wont really be possible given we control the token whitelisting process.

what additional checks do you think would be useful to add for additional safety?